miner/worker.go

```diff
@@ -739,15 +742,29 @@ func (w *worker) commitTransaction(env *environment, tx *types.Transaction) ([]*
 		snap = env.state.Snapshot()
 		gp   = env.gasPool.Gas()
 	)
+
+	// Checking against blob gas limit: It's kind of ugly to perform this check here, but there
+	// isn't really a better place right now. The blob gas limit is checked at block validation time
+	// and not during execution. This means core.ApplyTransaction will not return an error if the
+	// tx has too many blobs. So we have to explicitly check it here.
+	if (env.blobs+len(tx.BlobHashes()))*params.BlobTxBlobGasPerBlob > params.MaxBlobGasPerBlock {
```

Comment:
I think it can be treated as `block.GasLeft`.

Block has a gasPool and we can create a blobGasPool for blobTx. The benefit of this approach is: this check can be used in both (1) block generation and (2) block import.

Right now, we check the blob validity in `(v *BlockValidator) ValidateBody` when import a new block, and here when generate a new block. They can be in the same place I think.

Reply:
Me and @lightclient tried making this change yesterday, adding blobGas into core.GasPool. I think it's not worth it right now. There are too many places in the code where we would need to suddenly care about blobGas. It would be better to refactor the state transition first, to remove duplication across the codebase.